drivers/SmartThings/zwave-switch/src/multi-metering-switch/init.lua

```diff
@@ -0,0 +1,168 @@
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }), component)
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }), component)
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local children_amount = MULTI_METERING_SWITCH_CONFIGURATION_MAP.get_child_amount(device)
+    for i = 2, children_amount+1, 1 do
+      local device_name_without_number = string.sub(driver.label, 0,-2)
+      local name = string.format("%s%d", device_name_without_number, i)
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = MULTI_METERING_SWITCH_CONFIGURATION_MAP.get_child_switch_device_profile(device),
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
+  end
+  do_refresh(driver, device)
+end
+
+local function find_child(parent, ep_id)
+  if ep_id == PARENT_ENDPOINT then
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+  end
+end
+
+local function component_to_endpoint(device, component)
+  return PARENT_ENDPOINT
+end
+
+local function device_init(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function switch_binary_report_handler(driver, device, cmd)
+  local event
+  local newValue
+  if (cmd.args.target_value ~= nil) then
+    newValue = cmd.args.target_value
+  elseif cmd.args.value ~= nil then
+    newValue = cmd.args.value
+  end
+  
+  if newValue ~= nil and cmd.src_channel > 0 then
+    if newValue == SwitchBinary.value.OFF_DISABLE then
+      event = capabilities.switch.switch.off()
+    else
+      event = capabilities.switch.switch.on()
+    end
+    device:emit_event_for_endpoint(cmd.src_channel, event)
+  end
+end
+
+local map_unit = {
+  [Meter.scale.electric_meter.WATTS] = "W",
+  [Meter.scale.electric_meter.KILOWATT_HOURS] = "kWh"
+}
+
+local map_scale_to_capability = {
+  [Meter.scale.electric_meter.WATTS] = capabilities.powerMeter.power,
+  [Meter.scale.electric_meter.KILOWATT_HOURS] = capabilities.energyMeter.energy,
+}
+
+local function power_energy_meter_report_handler(self, device, cmd)
+  local supportedUnit = map_unit[cmd.args.scale]
+  
+  if cmd.src_channel > 0 and supportedUnit ~=nil then
+    local event_arguments = {
+      value = cmd.args.meter_value,
+      unit = supportedUnit
+    }
+    
+    local capabilityAttribute = map_scale_to_capability[cmd.args.scale]
+    device:emit_event_for_endpoint(
+      cmd.src_channel,
+      capabilityAttribute(event_arguments)
+    )
+  end
+end
```

Comment:
The defaults don't work here because you have to specifically filter out events from channel `0`, right?

I'd still prefer if you called the default handlers from within this method, though. After you filter by channel, you should be able to pass to the Power and Energy defaults like we do here: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zwave-lock/src/keywe-lock/init.lua#L57

Since they both have built-in unit checking, you can just pass the `cmd` without checking the unit. That should also allow you to delete those maps.

Reply:
Ok, I deleted these methods, so now it is handled by defaults